crates/exex/exex/src/backfill.rs

```diff
@@ -0,0 +1,355 @@
+use reth_db_api::database::Database;
+use reth_evm::execute::{BatchExecutor, BlockExecutionError, BlockExecutorProvider};
+use reth_node_api::FullNodeComponents;
+use reth_primitives::{Block, BlockNumber};
+use reth_provider::{
+    Chain, FullProvider, HistoricalStateProviderRef, ProviderError, TransactionVariant,
+};
+use reth_prune_types::PruneModes;
+use reth_revm::database::StateProviderDatabase;
+use reth_stages_api::{format_gas_throughput, ExecutionStageThresholds};
+use reth_tracing::tracing::{debug, trace};
+use std::{
+    marker::PhantomData,
+    ops::RangeInclusive,
+    time::{Duration, Instant},
+};
+
+/// Factory for creating new backfill jobs.
+#[derive(Debug, Clone)]
+pub struct BackfillJobFactory<E, P> {
+    executor: E,
+    provider: P,
+    prune_modes: PruneModes,
```

Comment:
do we even need these?

Reply:
yeah, for the batch executor, but it feels out of place in the executor API tbh

Reply:
same, I never understood why this is even in there tbh